### PR TITLE
glrnvim: init at 1.4.0

### PR DESCRIPTION
**Pull request by aacebedo**
## Description of changes

Add this tool that helps with the integration of nvim as a default editor in a system. 

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/gl/glrnvim/package.nix [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+  cargoSha256 = "sha256-cHEse+pXwgPTL8GJyY4s1mhWXGTY8Fnn2rFpA5SNerY=";
+
+  postInstall = ''
+    install -Dm644 glrnvim.desktop -t $out/share/applications
+    install -Dm644 glrnvim.svg $out/share/icons/hicolor/scalable/apps/glrnvim.svg
+  '';
+
+  meta = {
+    description = "A really fast & stable neovim GUI could be accelerated by GPU.";
```
Comment (SuperSandro2000) [inline pkgs/by-name/gl/glrnvim/package.nix:27]:
```suggestion
    description = "A really fast & stable neovim GUI could be accelerated by GPU";
```

File: pkgs/by-name/gl/glrnvim/package.nix [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+  rustPlatform,
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "glrnvim";
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "beeender";
+    repo = pname;
```
Comment (SuperSandro2000) [inline pkgs/by-name/gl/glrnvim/package.nix:14]:
```suggestion
    repo = "glrnvim";
```

Comment (aacebedo) [reply pkgs/by-name/gl/glrnvim/package.nix:14]: done

File: pkgs/by-name/gl/glrnvim/package.nix (hunk 1 above)
Comment (aacebedo) [reply pkgs/by-name/gl/glrnvim/package.nix]: done

File: pkgs/by-name/gl/glrnvim/package.nix [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  cargoSha256 = "sha256-cHEse+pXwgPTL8GJyY4s1mhWXGTY8Fnn2rFpA5SNerY=";
+
+  postInstall = ''
+    install -Dm644 glrnvim.desktop -t $out/share/applications
+    install -Dm644 glrnvim.svg $out/share/icons/hicolor/scalable/apps/glrnvim.svg
+  '';
+
+  meta = {
+    description = "A really fast & stable neovim GUI could be accelerated by GPU";
```
Comment (SuperSandro2000) [inline pkgs/by-name/gl/glrnvim/package.nix:27]: There was recently a tree wide change which fixed many meta.descriptions to not contain a leading The or A. Please drop that to keep the codebase in a clean state.

File: pkgs/by-name/gl/glrnvim/package.nix [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    license = lib.licenses.gpl3Plus;
+    maintainers = with lib.maintainers; [ aacebedo ];
+    platforms = lib.platforms.unix;
```
Comment (SuperSandro2000) [inline pkgs/by-name/gl/glrnvim/package.nix:30]: unix is the default platform for buildRustPackage. So we can commit that.

File: pkgs/by-name/gl/glrnvim/package.nix [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  version = "1.4.0";
+
+  src = fetchFromGitHub {
+    owner = "beeender";
+    repo = "glrnvim";
+    rev = "v${version}";
+    sha256 = "sha256-n3t3s3fzmBGXRDydYxNJ13itKul8dyLNW6HP8Di4hY0=";
+  };
+
+  cargoSha256 = "sha256-cHEse+pXwgPTL8GJyY4s1mhWXGTY8Fnn2rFpA5SNerY=";
```
Comment (SuperSandro2000) [inline pkgs/by-name/gl/glrnvim/package.nix:19]: We should use the newer cargoHash here

Comment (aacebedo) [reply pkgs/by-name/gl/glrnvim/package.nix:19]: fixed

File: pkgs/by-name/gl/glrnvim/package.nix (hunk 4 above)
Comment (aacebedo) [reply pkgs/by-name/gl/glrnvim/package.nix:30]: remove

File: pkgs/by-name/gl/glrnvim/package.nix (hunk 3 above)
Comment (aacebedo) [reply pkgs/by-name/gl/glrnvim/package.nix:27]: removed

**Comment by aacebedo** [comment]
@SuperSandro2000 can you merge this if the modifications are correct?

File: pkgs/by-name/gl/glrnvim/package.nix [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+
+  postInstall = ''
+    install -Dm644 glrnvim.desktop -t $out/share/applications
+    install -Dm644 glrnvim.svg $out/share/icons/hicolor/scalable/apps/glrnvim.svg
+  '';
+
+  meta = {
+    description = "really fast & stable neovim GUI could be accelerated by GPU";
+    license = lib.licenses.gpl3Plus;
+    maintainers = with lib.maintainers; [ aacebedo ];
```
Comment (Aleksanaa) [inline pkgs/by-name/gl/glrnvim/package.nix:29]:
```suggestion
    maintainers = with lib.maintainers; [ aacebedo ];
```

Please add `mainProgram`

File: pkgs/by-name/gl/glrnvim/package.nix [hunk 7]
```diff
@@ -0,0 +1,31 @@
...
+
+  cargoHash = "sha256-cHEse+pXwgPTL8GJyY4s1mhWXGTY8Fnn2rFpA5SNerY=";
+
+  postInstall = ''
+    install -Dm644 glrnvim.desktop -t $out/share/applications
+    install -Dm644 glrnvim.svg $out/share/icons/hicolor/scalable/apps/glrnvim.svg
+  '';
+
+  meta = {
+    description = "really fast & stable neovim GUI could be accelerated by GPU";
```
Comment (Aleksanaa) [inline pkgs/by-name/gl/glrnvim/package.nix:27]: Please add `homepage`. If there's no dedicated homepage, use GitHub repo page.

File: pkgs/by-name/gl/glrnvim/package.nix (hunk 6 above)
Comment (aacebedo) [reply pkgs/by-name/gl/glrnvim/package.nix:29]: done

File: pkgs/by-name/gl/glrnvim/package.nix (hunk 7 above)
Comment (aacebedo) [reply pkgs/by-name/gl/glrnvim/package.nix:27]: done

File: pkgs/by-name/gl/glrnvim/package.nix [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
```
Comment (Aleksanaa) [inline pkgs/by-name/gl/glrnvim/package.nix:27]:
```suggestion
    description = "Really fast & stable neovim GUI which could be accelerated by GPU";
```

Comment (aacebedo) [reply pkgs/by-name/gl/glrnvim/package.nix]: corrected

**Comment by aacebedo** [comment]
@Aleksanaa can you merge?

File: pkgs/by-name/gl/glrnvim/package.nix (hunk 8 above)
Comment (Aleksanaa) [reply pkgs/by-name/gl/glrnvim/package.nix]: The syntax here doesn't look quite right, is it?

Comment (aacebedo) [reply pkgs/by-name/gl/glrnvim/package.nix]: corrected

**Comment by aacebedo** [comment]
@Aleksanaa is it ok for you now I have corrected the comment?

**Review by Aleksanaa — APPROVED** [review]

**Comment by superherointj** [comment]
Builds for Darwin are failing:

* https://hydra.nixos.org/build/264826359
* https://hydra.nixos.org/build/264825891

**Comment by Aleksanaa** [comment]
Sorry, I did a bunch of shit last night. I wasn't wide awake.

**Comment by superherointj** [comment]
> Sorry, I did a bunch of shit last night. I wasn't wide awake. 

It is too many details that can go wrong. Thanks for still trying though. Let's just fix this then.

**Comment by superherointj** [comment]
> Sorry, I did a bunch of shit last night. I wasn't wide awake. 

Please, don't feel sorry.

It is too many details that can go wrong.
You have provided your time and effort in reviewing/merging this. Thanks for it.
Let's just fix this detail.

[By fix I mean any fix I listed, exception being having hydra to continuously build this. Which I believe is the worst option of all, long term at least.]

**Comment by aacebedo** [comment]
Sorry too. I missed the error message.

**Pull request merged**